apisix/plugins/traffic-split.lua

```diff
@@ -200,7 +200,7 @@ local function new_rr_obj(weighted_upstreams)
     for i, upstream_obj in ipairs(weighted_upstreams) do
         if upstream_obj.upstream_id then
             server_list[upstream_obj.upstream_id] = upstream_obj.weight
-        elseif upstream_obj.upstream then
+        elseif upstream_obj.upstream and type(upstream_obj.upstream) == "table" then
```

Comment:
It doesn't fix the user's problem, the `upstream` property shouldn't be a string, if we add the judgement here, it will cause other problem elsewhere

Reply:
@monkeyDluffy6017 Yes I agree with you but the user's bug wasn't reproducible and the data gets corrupted later on so adding this extra check just avoid that nil panic, and nothing else.

Reply:
We cannot just add the check here,  The `upstream_obj.upstream` is used in other parts of the code as well

Reply:
so should we not fix this on apisix's end since this data was not potentially corrupted by APISIX? Should I close this PR?

Reply:
Please close this pr and find the root cause of the problem